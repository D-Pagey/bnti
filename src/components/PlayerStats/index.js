import React from 'react';
import { string, number } from 'prop-types';

const PlayerStats = ({ playerName, kills, deaths, wins }) => {
    return (
        <div>
            <h3>{`${playerName}'s stats`}</h3>
            <ul>
                <li>Kills: {kills}</li>
                <li>Deaths: {deaths}</li>
                <li>Wins: {wins}</li>
            </ul>
        </div>
    );
};

PlayerStats.propTypes = {
    playerName: string.isRequired,
    kills: number.isRequired,
    deaths: number.isRequired,
    wins: number.isRequired
};

export default PlayerStats;
