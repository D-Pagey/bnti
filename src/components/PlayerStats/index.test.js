import React from 'react';
import { render } from 'react-testing-library';

import PlayerStats from '.';

const props = {
    kills: 500,
    deaths: 250,
    wins: 1,
    playerName: 'GingerVking'
};

describe('PlayerStats component', () => {
    it('should render', () => {
        const { container } = render(<PlayerStats {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
