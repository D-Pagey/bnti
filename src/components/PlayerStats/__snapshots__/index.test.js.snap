// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerStats component should render 1`] = `
<div>
  <h3>
    GingerVking's stats
  </h3>
  <ul>
    <li>
      Kills: 
      500
    </li>
    <li>
      Deaths: 
      250
    </li>
    <li>
      Wins: 
      1
    </li>
  </ul>
</div>
`;
